Parsers - Logstash changes and new CommmonSecurityLogs AdditionalExtension Parser

ashwin-patil opened the pull request:
Fixes #

## Proposed Changes

  - Updated references to new official Logstash plugin in existing parser config files
  - New CommonSecurityLog parser to convert AdditionalExtensions into Json.

shainw commented on Parsers/Logstash/input-jsonfile-output-loganalytics.conf:
```diff
@@ -1,8 +1,11 @@
 // Logstash configuration file to ingest custom json file
-// Last Updated Date: Nov 5, 2019
+// Last Updated Date: Aug 14, 2019
```
I assume you mean Aug 14th, 2020. ;)

shainw commented on Parsers/Logstash/input-aws_s3-output-loganalytics.conf:
```diff
@@ -1,8 +1,11 @@
-// Logstash configuration file to ingest custom json file
-// Last Updated Date: Nov 5, 2019
+// Logstash configuration file to ingest files from AWS S3 buckets
+// Last Updated Date: Aug 14, 2019
```
I assume you mean Aug 14th, 2020. ;)

shainw commented on Parsers/CommonSecurityLogs-AdditionalExtensionParser.txt:
```diff
@@ -0,0 +1,23 @@
+//
+// Parser Notes: 
+// 1. The AdditionalExtensions field generally captures additional fields which are not parsed/mapped to fields by respective vendor CEF configurations
+// 2. The field will have dynamic set of key value pairs separated by ; in the format of the key-balue;key=value;key=value
```
Should be key=value?

shainw commented on Parsers/CommonSecurityLogs-AdditionalExtensionParser.txt:
```diff
@@ -0,0 +1,23 @@
+// 1. The AdditionalExtensions field generally captures additional fields which are not parsed/mapped to fields by respective vendor CEF configurations
+// 2. The field will have dynamic set of key value pairs separated by ; in the format of the key-balue;key=value;key=value
+// e.g. AdditionalExtensions = deviceTranslatedPort=60095;tunnelType=IPSEC;dnat=No
+// 3. The function takes these dynamic key value pairs and converts into json for easier accessing.
```
"converts them into json".<stop>  no need to include the "for easier accessing" part.

ashwin-patil commented:
> Now working, apologies for hijacking this thread.
> `| project SyslogMessage = extract_all(@"(?P<key>[A-Za-z0-9.]*)=(?P<value>[A-Za-z0-9.]*)", dynamic(["key","value"]), SyslogMessage), TimeGenerated`

Thank you for reporting. I have modified to regex to support parsing for IP, hostnames etc and raised new PR #1014 with the fix included.

Modified parser looks like below
```
CommonSecurityLog
| extend AdditionalExtensions = extract_all(@"(?P<key>\w+)=(?P<value>[a-zA-Z0-9-_.]+)", dynamic(["key","value"]), AdditionalExtensions)
| mv-apply AdditionalExtensions on (
    summarize AdditionalExtensionsParsed = make_bag(pack(tostring(AdditionalExtensions[0]), AdditionalExtensions[1]))
)
```
Sample screenshot showing IP, hostname getting parsed

![image](https://user-images.githubusercontent.com/11397335/91214618-2f526c80-e6c8-11ea-91f7-096e56775d2f.png)
